rules/group.go

```diff
@@ -84,6 +95,11 @@ type Group struct {
 // DefaultEvalIterationFunc is the default implementation.
 type GroupEvalIterationFunc func(ctx context.Context, g *Group, evalTimestamp time.Time)
 
+// EvaluateRuleGroup will be used to determine if a rule group
+// needs to be evaluated or deactivated. Deactivating a rule group will
+// reset the metrics but not unload the rule group.
+type EvaluateRuleGroup func(ctx context.Context, g *Group, evalTimestamp time.Time, doneCh <-chan struct{}) bool
```

Comment:
Nit: I think this needs a better name because `EvaluateRuleGroup` sounds like a function that performs that evaluation, not a functions that decides whether to evaluate or not. 
Suggestion: `RuleGroupEvalControlFunc`